commit/merkleroot/rmn/client.go

```diff
@@ -19,13 +41,595 @@ type Client interface {
 	ComputeReportSignatures(
 		ctx context.Context,
 		destChain *rmnpb.LaneDest,
-		requestedUpdates []rmnpb.FixedDestLaneUpdateRequest,
+		requestedUpdates []*rmnpb.FixedDestLaneUpdateRequest,
 	) (*ReportSignatures, error)
 }
 
 type ReportSignatures struct {
-	// ReportSignatures are the ECDSA signatures for the lane updates for each node.
+	// Signatures are the ECDSA signatures for the lane updates for each node.
+	Signatures []*rmnpb.EcdsaSignature
+	// LaneUpdates are the lane updates for each source chain that we got the signatures for.
 	// NOTE: A signature[i] corresponds to the whole LaneUpdates slice and NOT LaneUpdates[i].
-	Signatures  []rmnpb.EcdsaSignature
-	LaneUpdates []rmnpb.FixedDestLaneUpdate
+	LaneUpdates []*rmnpb.FixedDestLaneUpdate
+}
+
+// client is the base RMN Client implementation.
+type client struct {
+	lggr                                    logger.Logger
+	rawRmnClient                            RawRmnClient
+	rmnNodes                                []RMNNodeInfo
+	rmnRemoteAddress                        cciptypes.Bytes
+	rmnHomeConfigDigest                     cciptypes.Bytes
+	minObservers                            int
+	minSigners                              int
+	observationsInitialRequestTimerDuration time.Duration
+	reportsInitialRequestTimerDuration      time.Duration
+}
+
+// RMNNodeInfo contains the information about an RMN node.
+type RMNNodeInfo struct {
+	// ID is the index of this node in the RMN config
+	ID                    uint32
```

Comment:
Very nitty: would be nice to have a type alias of this `type RMNNodeID = uint32` and can use that in all the types below to be a bit more expressive.